extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.Optimizely.js

```diff
@@ -0,0 +1,19 @@
+/*!
+ * Utility functions for running Optimizely A/B tests with VisualEditor
+ */
+
+( function () {
+	/**
+	 * @returns {number} The A/B test variant number for the Toolbar A/B test
+	 */
+	ve.init.wikia.getToolbarABTestVariantNumber = function () {
+		var optimizely = window.optimizely,
+			optimizelyId = window.wgDevelEnvironment ? '4721410313' : '4701112678';
+
+		if (optimizely && optimizely.variationMap && optimizely.variationMap.hasOwnProperty(optimizelyId)) {
+			return optimizely.variationMap[optimizelyId] || 0;
```

Comment:
Everything on optimizely.data.state is also replicated directly on the optimizely object. I can't find this explicitly in the documentation, but optimizely.variationMap is used directly throughout Mercury.

I know it is possible to inject javascript too, but I was explicitly advised to just use variationMap by the Mercury team to avoid polluting the global namespace.